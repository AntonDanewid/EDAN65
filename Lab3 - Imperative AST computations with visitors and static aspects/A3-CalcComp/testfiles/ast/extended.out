Program
  List
    FuncDecl
      IdDecl
      List
      Block
        List
          WhileStmt
            NE
              IdUseExpr
              IdUseExpr
            Block
              List
                IfStmt
                  GT
                    IdUseExpr
                    IdUseExpr
                  Block
                    List
                      AssignStmt
                        IdUseExpr
                        Sub
                          IdUseExpr
                          IdUseExpr
                  Opt
                    Block
                      List
                        AssignStmt
                          IdUseExpr
                          Sub
                            IdUseExpr
                            IdUseExpr
          ReturnStmt
            IdUseExpr
    FuncDecl
      IdDecl
      List
        IdDecl
        IdDecl
      Block
        List
    FuncDecl
      IdDecl
      List
        IdDecl
        IdDecl
      Block
        List
          DeclStmt
            IdDecl
          DeclStmt
            IdDecl
          AssignStmt
            IdUseExpr
            ReadFunc
              Opt
          AssignStmt
            IdUseExpr
            ReadFunc
              Opt
          PrintStmt
            FuncUseExpr
              IdUseExpr
              List
                IdUseExpr
                IdUseExpr
          PrintStmt
            FuncUseExpr
              IdUseExpr
              List
                IdUseExpr
                IdUseExpr
          ReturnStmt
            IntLiteral
