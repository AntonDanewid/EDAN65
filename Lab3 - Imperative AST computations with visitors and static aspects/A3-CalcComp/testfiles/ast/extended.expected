Program
  List
    FuncDecl
      IdDecl
      Opt
      Block
        Opt
          List
            WhileStmt
              NE
                IdUseExpr
                IdUseExpr
              Block
                Opt
                  List
                    IfStmt
                      GT
                        IdUseExpr
                        IdUseExpr
                      Block
                        Opt
                          List
                            AssignStmt
                              IdUseExpr
                              Sub
                                IdUseExpr
                                IdUseExpr
                      Opt
                        Block
                          Opt
                            List
                              AssignStmt
                                IdUseExpr
                                Sub
                                  IdUseExpr
                                  IdUseExpr
            ReturnStmt
              IdUseExpr
    FuncDecl
      IdDecl
      Opt
        List
          IdDecl
          IdDecl
      Block
        Opt
    FuncDecl
      IdDecl
      Opt
        List
          IdDecl
          IdDecl
      Block
        Opt
          List
            DeclStmt
              IdDecl
            DeclStmt
              IdDecl
            AssignStmt
              IdUseExpr
              ReadFunc
            AssignStmt
              IdUseExpr
              ReadFunc
            PrintStmt
              FuncUseExpr
                IdUseExpr
                List
                  IdUseExpr
                  IdUseExpr
            PrintStmt
              FuncUseExpr
                IdUseExpr
                List
                  IdUseExpr
                  IdUseExpr
            ReturnStmt
              IntLiteral
